tests/unit/states/test_module.py

```diff
@@ -128,6 +128,34 @@ def test_run_module_not_available(self):
         if ret["comment"] != "Unavailable function: {}.".format(CMD) or ret["result"]:
             self.fail("module.run did not fail as expected: {}".format(ret))
 
+    def test_run_module_not_available_testmode(self):
+        """
+        Tests the return of module.run state when the module function is not available
+        when run with test=True
+        :return:
+        """
+        with patch.dict(module.__salt__, {}, clear=True), patch.dict(
+            module.__opts__, {"test": True, "use_superseded": ["module.run"]}
+        ):
+            ret = module.run(**{CMD: None})
+        if (
+            ret["comment"] != "Unavailable function: {}.".format(CMD)
+            or ret["result"] is not False
+        ):
+            self.fail("module.run did not fail as expected: {}".format(ret))
+
+    def test_run_module_noop(self):
+        """
+        Tests the return of module.run state when no module function is provided
+        :return:
+        """
+        with patch.dict(module.__salt__, {}, clear=True), patch.dict(
+            module.__opts__, {"test": True, "use_superseded": ["module.run"]}
+        ):
+            ret = module.run(**{})
```

Comment:
```suggestion
            ret = module.run()
```
Is `**{}` really necessary?

Reply:
Good catch. I have removed that.